src/shims/unix/fs.rs

```diff
@@ -639,7 +639,8 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         &mut self,
         path_op: &OpTy<'tcx>,
         buf_op: &OpTy<'tcx>,
-    ) -> InterpResult<'tcx, Scalar> {
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx, EmulateItemResult> {
```

Comment:
Is it really worth having the `EmulateItemResult` everywhere? What's the problem with the previous approach here?

Reply:
consistency. If we want all FFI shims to work the same way and generate all of them with a macro, we either need to do this or have some trait shenanigans that allow `()` or `EmulateItemResult` and treating `()` as the usual case

Reply:
We can have the macro support "simple" shims and "destination passing style" shims, to avoid overcomplicating the extremely common case.